packages/core/_test/stm/TReentrantLock/locks.test.ts

```diff
@@ -0,0 +1,234 @@
+const pollSchedule = <E, A>(): Schedule<Tuple<[number, void]>, never, Option<Exit<E, A>>, Option<Exit<E, A>>> =>
+  (Schedule.recurs(100) > Schedule.identity<Option<Exit<E, A>>>()).whileOutput((_) => _.isNone())
+
+describe.concurrent("TReentrantLock", () => {
+  describe.concurrent("locks", () => {
+    it("1 read lock", async () => {
+      const program = Do(($) => {
+        const lock = $(TReentrantLock.make().commit())
+        const count = $(Effect.scoped(lock.readLock().flatMap((count) => Effect.succeed(count))))
+
+        return count === 1
+      })
+
+      const result = await program.unsafeRunPromise()
+
+      assert.isTrue(result)
+    })
+    it("2 read locks from same fiber", async () => {
+      const program = Do(($) => {
+        const lock = $(TReentrantLock.make().commit())
+        const count = $(
+          Effect.scoped(
+            lock.readLock().flatMap(() => Effect.scoped(lock.readLock().flatMap((count) => Effect.succeed(count))))
+          )
+        )
+
+        return count === 2
+      })
+
+      const result = await program.unsafeRunPromise()
+
+      assert.isTrue(result)
+    })
+    it("2 read locks from different fibers", async () => {
+      const program = Do(($) => {
+        const lock = $(TReentrantLock.make().commit())
+        const rlatch = $(Deferred.make<never, void>())
+        const mlatch = $(Deferred.make<never, void>())
+        const wlatch = $(Deferred.make<never, void>())
+
+        $(
+          Effect.scoped(
+            lock.readLock().flatMap((count) => mlatch.succeed(undefined as void) > rlatch.await().as(count))
+          ).fork()
+        )
+
+        $(mlatch.await())
+
+        const reader2 = $(
+          Effect.scoped(lock.readLock().flatMap((count) => wlatch.succeed(undefined as void).as(count))).fork()
+        )
+
+        $(wlatch.await())
+
+        const count = $(reader2.join())
+
+        return count === 1
+      })
+
+      const result = await program.unsafeRunPromise()
+
+      assert.isTrue(result)
+    })
+    it("1 write lock then 1 read lock, different fibers", async () => {
```

Comment:
@mikearnaldi or @IMax153 This test only has a difference with ZIO on the return value of option. Where ZIO expects it to be `Option.None`, whereas I am just getting a `Option.Some(Exit.Success(1))`. 

(https://github.com/zio/zio/blob/a822917ad730910fd69f38c341ab134995cfae95/core-tests/shared/src/test/scala/zio/stm/TReentrantLockSpec.scala#L57)

I couldn't really find a tangible reason why'd it would make sense for it to be a `Option.None`.